### PR TITLE
Added TypesScript support

### LINKED ISSUE #380: TypeScript support
https://github.com/DefinitelyTyped/DefinitelyTyped/search?utf8=%E2%9C%93&q=rc-calendar&type=

### PR DESCRIPTION
**Pull request by 9renpoto**
Closes #380

### CODE REVIEW

File: types/rc-calendar/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import * as React from 'react';
+import { Moment } from 'moment';
+
+export type Mode = 'time' | 'date' | 'month' | 'year' | 'decade';
+
+export interface Props {
+    prefixCls?: string;
+    className?: string;
+    style?: object;
```
Comment (yesmeck) [inline types/rc-calendar/index.d.ts]: `React.CSSProperties`

File: types/rc-calendar/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    locale?: object;
+    showDateInput?: boolean;
+    showWeekNumber?: boolean;
+    showToday?: boolean;
+    showOk?: boolean;
+    onSelect?: (date: Moment) => void;
+    onOk?: () => void;
+    onKeyDown?: () => void;
+    timePicker?: React.ReactNode;
+    dateInputPlaceholder?: any;
```
Comment (yesmeck) [inline types/rc-calendar/index.d.ts]: `string`?

File: types/rc-calendar/rc-calendar-tests.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import * as moment from 'moment';
+import Calendar from 'rc-calendar';
+
+const action = (date: moment.Moment) => {
+    date.subtract(1);
```
Comment (yesmeck) [inline types/rc-calendar/rc-calendar-tests.tsx]: Two whitespace indent.

File: types/rc-calendar/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
```
Comment (yesmeck) [inline types/rc-calendar/tsconfig.json]: Two whitespace indent.

File: types/rc-calendar/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+// Type definitions for rc-calendar 9.6
```
Comment (yesmeck) [inline types/rc-calendar/index.d.ts]: I'd like to put this file under the root of the project.

**Comment by yesmeck** [comment]
Need to add the `.d.ts` file to the `files` field of package.json.

File: types/rc-calendar/index.d.ts (hunk 5 above)
Comment (9renpoto) [reply types/rc-calendar/index.d.ts]: @yesmeck Thank you review 👍 

```console
        renamed:    types/rc-calendar/index.d.ts -> index.d.ts
        renamed:    types/rc-calendar/rc-calendar-tests.tsx -> rc-calendar-tests.tsx
        renamed:    types/rc-calendar/tsconfig.json -> tsconfig.json
        renamed:    types/rc-calendar/tslint.json -> tslint.json
```

Do I need to move the file as follows?
If that's okay add the path to package.json `files`

Comment (yesmeck) [reply types/rc-calendar/index.d.ts]: Yes, please just put these files under the root of the project.

**Comment by 9renpoto** [comment]
Fixed 👍

**Comment by 9renpoto** [comment]
Thanks !

**Pull request merged**